app/models/discourse_activity_pub_actor.rb

```diff
@@ -43,9 +41,17 @@ def keypair
   end
 
   def following?(actor)
+    return false unless actor
     actor.followers.exists?(id: self.id)
   end
 
+  def can_follow?(actor)
```

Comment:
Do we explicitly disallow a local actor to follow another local actor? In a federation context, this would make sense of course, but in Discourse, it feels odd to allow category A to "follow" category B, it would end up in needlessly duplicated content, right?

Reply:
Not necessarily. When an actor receives an activity they don't have to necessarily perform the activity itself. For example, we could relatively easily replicate the functionality of the [Follow Plugin ](https://meta.discourse.org/t/discourse-follow/110579) by allowing local actors to follow other local actors. It would just be a matter of adding the right notifications in the AP handlers in the plugin.rb and then a frontend in the user's profile (where you'd pull activities associated with actors you're following). But it's simpler if we restrict that for now, hence the restriction in the follow handler.